diff --git a/src/common/error.cpp b/src/common/error.cpp
--- a/src/common/error.cpp
+++ b/src/common/error.cpp
@@ -35,7 +35,7 @@ std::string NativeErrorToString(int e) {
 #else
     char err_str[255];
 #if defined(__GLIBC__) && (_GNU_SOURCE || (_POSIX_C_SOURCE < 200112L && _XOPEN_SOURCE < 600)) ||   \
-    (defined(ANDROID) && !defined(HAVE_LIBRETRO))
+    (defined(__ANDROID__) && defined(__USE_GNU) && __ANDROID_API__ >= 23)
     // Thread safe (GNU-specific)
     const char* str = strerror_r(e, err_str, sizeof(err_str));
     return std::string(str);
